apple/RNCWebView.m

```diff
@@ -973,24 +975,42 @@ - (void)                    webView:(WKWebView *)webView
   decidePolicyForNavigationResponse:(WKNavigationResponse *)navigationResponse
                     decisionHandler:(void (^)(WKNavigationResponsePolicy))decisionHandler
 {
+  WKNavigationResponsePolicy policy = WKNavigationResponsePolicyAllow;
   if (_onHttpError && navigationResponse.forMainFrame) {
```

Comment:
@trcoffman It seems like access to this functionality requires that `_onHttpError` is set. Is there a reason for that?

Reply:
So, you are right that this should be improved for the sake of readability. However, from taking another look at the code, it looks like _onHttpError will always be set whether or not the user provides an implementation, there is a default implementation that does nothing whether or not the user supplies one.

So this doesn't actually cause any functionality problems in the codebase.

Originally this `decidePolicyForNavigationResponse` was only used for the sake of gathering error codes to send to `onHttpError`. However, I added another use-case to this method and I should have moved the check for _onHttpError to surround just the relevant part.

Another improvement that should be made is that the entire section where we check the content-disposition header should be inside a check to see if _onFileDownload exists. This way unnecessary code isn't running for clients that don't supply _onFileDownload. Unlike _onHttpError, _onFileDownload should actually be null if the user doesn't supply an implementation. This is such an easy optimization that I'm kicking myself for not writing it that way to begin with. It seems so obvious now that I give the code a fresh look.

I'm not really sure why default implementations like this in `WebView.ios.tsx` exist:
```
  onHttpError = (event: WebViewHttpErrorEvent) => {
    const { onHttpError } = this.props;
    if (onHttpError) {
      onHttpError(event);
    }
  }
```

All this does is ensure that _onHttpError will never be null in the native code even if the user doesn't supply a method. Additional code was written which just results in useless CPU cycles for users that don't supply an onHttpError handler. In library code for power constrained mobile devices, when we can optimize something simply by deleting code that seems like a win-win. Less code to maintain and it is more efficient.